File: 334_IncreasingTripletSubsequence.cpp
//35.2%
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        if(nums.size() < 3)
			return false;
		int s1(INT_MAX), s2(INT_MAX);
		for(int num : nums){
			if(num <= s1){	
				s1 = num;
			}else if(num <= s2)
				s2 = num;
			else
				return true;
		}
		return false;
    }
};

File: 334_IncreasingTripletSubsequence.py
##37.74%
import sys
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        if len(nums) < 3:
            return False
        s1, s2 = nums[0], sys.maxint
        for num in nums:
            if num <= s1:
                s1 = num
            elif num <= s2:
                s2 = num
            else:
                return True
        return False
		
def increasingTriplet(nums):
    first = second = float('inf')
    for n in nums:
        if n <= first:
            first = n
        elif n <= second:
            second = n
        else:
            return True
    return False